### PR TITLE
[diffusion] Bit-exact data-movement elimination for the Wan causal VAE decoder (H200 LongLive2 704x1280x61f: decode 2.80->2.32 s lossless / 2.12->1.67 s quality=high, e2e -10.7%)

### PR DESCRIPTION
**Pull request by BBuf**
Branch: `p2-wan-vae-decode`, single commit rebased onto current `main`, standalone — no dependencies (rebase was clean; unit tests re-verified green on current `main`). 3 files, +666/−111 (new Triton module, wanvae rewiring, registered unit test). Complements the merged #33546 (quality-gated Wan RMSNorm+SiLU): #33546 accelerates the pointwise norm chain behind `quality=high`; this PR removes the decoder's data-movement tax on the **default lossless path** — every change is pure data movement plus zero fill (and one fp32-accumulated add), so no gate is needed and golden outputs are unchanged (md5-verified end to end).

## Motivation

On few-step Wan-family video pipelines the causal VAE decode is the dominant fixed cost: LongLive2 (704x1280, 61 frames, 4-step DMD, H200 bf16) spends 2.8 s of a 4.5 s pipeline in DecodingStage = **62% of e2e**; FastWan2.2-TI2V-5B (3 steps, fp32 decode — the Wan2.2 default) spends 4.2 s of 5.2 s = **81%**.

A kernel-level attribution of the 2.59 s LongLive2 decode GPU time shows conv itself is only ~1.26 s. **0.51 s is `aten::copy_` (1849 calls) plus 60 ms `fill_` and 39 ms `cat`** — all bookkeeping around the causal feature cache and the channels_last_3d layout — and another ~0.28 s sits in the `DupUp3D` shortcut arm:

| source (per decode, trace-attributed, H200) | GPU ms |
|---|---|
| `F.pad` copy + zero fill inside `causal_conv3d_cat_pad` | 268 |
| `contiguous(channels_last_3d)` in `match_conv3d_input_format` | 108 |
| input repack inside `aten::convolution` (cat/pad output not NDHWC-clean) | 96 |
| feature-cache `clone`/`cat` upkeep + `torch.cat` of cache+hidden | 39 |
| `DupUp3D` `repeat_interleave` + `permute().contiguous()` | 71 |
| `x + avg_shortcut(x_copy)` separate add pass | 103 |

The per-conv pattern on every chunk is: `clone` the last two frames for the next cache entry, `cat` the previous cache with the hidden state, `F.pad` (copy + fill), then `contiguous(channels_last_3d)` for the NHWC conv — four full passes over every conv input, none doing arithmetic. The `DupUp3D` arm is worse than it looks: on the `upsample2d` up-block the tensors flow in NHWC-style layout, and the aten `repeat_interleave + contiguous + add` chain pays heavily for the mixed layouts (7.9 ms for the largest call — the fused kernel does it in 3.3 ms by following the output's memory order).

## Modifications

**`sglang/kernels/ops/diffusion/triton/wan_causal_cache.py` (new)** — two strided data-movement kernels:

- `cat_pad_channels_last_3d(x, cache, padding, keep_cache_t=)`: builds the causal conv input `contiguous_cl3d(F.pad(cat([cache, x], 2), padding))` in **one pass** (strided reads, NDHWC writes, zero fill in-kernel), and in the same pass optionally emits the **compact next-chunk feature cache** (unpadded interior of the last `CACHE_T` frames) as a second output — the clone/cat cache upkeep disappears entirely and the conv-input buffer is freed after the conv instead of being pinned.
- `dup_up3d_add(main, src, ...)`: `main + DupUp3D(src)` in one gather+add pass. The output is `empty_like(main)` (the layout the aten add produces, so downstream layout-sensitive reductions see identical memory formats) and the traversal order follows that layout (channels-innermost for the NHWC-style `upsample2d` arm) so stores and `main` loads stay coalesced. The pixel-shuffle factors are `constexpr` powers of two (shift/mask, no per-element integer division).

**`runtime/models/vaes/wanvae.py`** — the six feature-cache conv sites (residual block conv1/conv2, encoder/decoder conv_in/conv_out, WanResample time_conv including the `"Rep"` first-chunk marker) route through one helper, `_run_cached_causal_conv`; `WanCausalConv3d.forward` uses the fused builder for the cache-less padded case; `residual_up_block_forward` uses `dup_up3d_add`. Every site falls back to the verbatim original op chain when the kernel does not support the request (non-CUDA, spatial-parallel conv subclass, non-NDHWC weights, `torch.compile` tracing, device/dtype mismatch, asymmetric pads), and the compact cache holds exactly the reference cache values so fused and fallback chunks can interleave freely.

**Why this is bit-exact, not "close"**: `cat`→`pad` and pad-each-then-`cat` place identical values (zero fill commutes with concatenation); the conv consumes a tensor with identical bits, shape, layout, and cudnn algorithm; the DupUp3D arm is a gather plus one add of the same two operands in the same order with the same fp32 accumulation and the same output layout aten uses. Unit-tested bitwise at kernel level and md5-verified end to end.

## Speedup

All numbers H200, seed 42, `CUDA_VISIBLE_DEVICES` pinned, one warmed process, 10 sequential identical requests, first dropped, median of 9. e2e = sum of pipeline stage times (the client wall clock additionally contains a ~0.8 s frame-return path whose variance swamps stage-level deltas).

**LongLive2 (Rabinovich/LongLive-2.0-5B-Diffusers, 704x1280, 61 frames, 4-step DMD, bf16 decode):**

| arm | DecodingStage | pipeline e2e | peak GPU mem |
|---|---|---|---|
| main, lossless | 2.802 s | 4.513 s | 49.6 GB |
| this PR, lossless | **2.323 s (−17.1%)** | **4.031 s (−10.7%)** | **46.1 GB** |
| main, `--quality high` (#33546) | 2.120 s | 3.838 s | |
| this PR, `--quality high` | **1.669 s (−21.3%)** | **3.378 s (−12.0%)** | |

Relative to today's default (main lossless), this PR plus the existing `--quality high` tier takes the decode from 2.802 s to 1.669 s (**−40%**).

**FastWan2.2-TI2V-5B (FastVideo/FastWan2.2-TI2V-5B-FullAttn-Diffusers, 704x1280, 61 frames, 3 steps, fp32 decode — family spillover, same VAE):**

| arm | DecodingStage | pipeline e2e |
|---|---|---|
| main, lossless | 4.243 s | 5.235 s |
| this PR, lossless | **3.614 s (−14.8%)** | **4.606 s (−12.0%)** |

**Attribution closure** (same trace tooling, profiled LongLive2 lossless decode window): `aten::copy_` 513 ms / 1849 calls → **151 ms / 708 calls**; `fill_` 60 → 0 ms; `cat` 39 → 6 ms; `DupUp3D`+shortcut-add 175 → 50 ms. The ~675 ms of removed aten movement is replaced by 155 ms of `_cat_pad_cl3d_kernel` (435 calls) + 50 ms of `_dup_up3d_add_kernel` (41 calls) — a net −470 ms of GPU work that matches the measured −479 ms decode delta; the window stays GPU-bound (99.4% busy) before and after, so the wall-clock gain is fully explained by the removed movement.

**Microbenchmarks** (H200, bf16, in-situ decoder shapes and strides): fused cat+pad ~3x over the clone+cat+pad+contiguous chain (e.g. C160 352x640: 0.47 vs 1.40 ms); `dup_up3d_add` 2.4-3.9x over the aten chain (largest `upsample2d` arm with the real NHWC/NTCHW layout mix: 3.32 vs 7.88 ms).

## Accuracy

- **Lossless default is bit-exact end-to-end**: frame-stream md5 identical to `origin/main` (52afe87a08) for LongLive2 (bf16) and FastWan2.2-TI2V-5B (fp32), 10/10 requests each (and md5 is request-stable within every arm).
- **`--quality high`**: byte-identical to main's `--quality high` output (same md5), i.e. this PR composes with #33546 without changing its numerics; PSNR vs the lossless reference is 58.6 dB (bar: >25 dB), unchanged from main.
- **Unit test**: `test/registered/kernels/ops/diffusion/test_wan_causal_cache.py` (registered CUDA CI): bitwise kernel-vs-aten equality across dtypes, shapes, and strided cache views (first-chunk zero fill, legacy 1-frame cache, encoder T=4, temporal-only pads, dual-output cache emit, `first_chunk` slicing, aten-matching output strides for the DupUp3D add), plus a chunked conv loop asserting the fused cache scheme reproduces the original clone/cat bookkeeping bitwise chunk by chunk, from both `None` and `"Rep"` starts. Existing `test_wan_vae_fastpath.py` still passes (21 passed total).

## Benchmark configuration

```
# per arm (PYTHONPATH switches main <-> this branch; add --quality high for the gated arms)
CUDA_VISIBLE_DEVICES=0 python p2_driver.py --num-requests 10 --out-json <arm>.json -- \
  --model-path Rabinovich/LongLive-2.0-5B-Diffusers \
  --prompt "A red fox running across a snowy meadow at sunrise, cinematic lighting" \
  --width 1280 --height 704 --num-frames 61 --seed 42
# family arm: --model-path FastVideo/FastWan2.2-TI2V-5B-FullAttn-Diffusers (same size/frames/seed)
```

Driver = one warmed `DiffGenerator` process looping `generate()`; frames returned in-process (`return_frames=True`, no video mux) and hashed streaming (md5 over raw frame bytes). LongLive-2.0-5B weights via the `Rabinovich/LongLive-2.0-5B-Diffusers` mirror.

## Checklist

- [x] Format your code according to the Format code with pre-commit.
- [x] Add unit tests.
- [x] Provide accuracy and speed benchmark results.
- [x] Follow the SGLang code style guidance.

---
### CI States

Latest PR Test (Base): :x: [Run #31269074863](https://github.com/sgl-project/sglang/actions/runs/31269074863)
Latest PR Test (Extra): :white_check_mark: [Run #31279010527](https://github.com/sgl-project/sglang/actions/runs/31279010527)

### CODE REVIEW

**Comment by BBuf** [comment]
Re-confirmed lossless is byte-exact on current `main` after rebase. LongLive2 704×1280×61f, 4-step DMD, seed 42, `quality=lossless`, H200: with vs without this PR produce an identical frame md5 `93468c316ad55b4aea26903629d98b91` (3/3 requests). Kernel-level bitwise equality vs the aten chain is covered by the registered unit test (22 passed on current main).

**Comment by BBuf** [comment]
CI is green on all required checks after re-running one shard: 14/14 `call-multimodal-gen-tests` lanes SUCCESS, plus `finish`, `lint`, `gate`, `check-changes` all SUCCESS.

The initial red on `multimodal-gen-test-1-gpu (2)` was a transient perf-threshold flake (all failing checks were latency assertions ~1–6% over, e.g. 693 vs 684 ms, on `joyai_image_edit` / `flux_2` / the Wan `InputValidationStage` — none on this PR's VAE-decode path; the identical shard passed on #34126 in the same runner pool), which fast-fail-cascaded the `base-b` lanes. Re-running the shard cleared it (now SUCCESS). Correctness is independently proven by the byte-exact lossless md5 above and the 22 bitwise kernel unit tests. Remaining reds are the known non-required `*-amd`/`*-npu`/`pr-test-*-finish` lanes (red on merged #34008 / #34085 too).
